src/main/java/org/alfresco/utility/data/DataContent.java

```diff
@@ -276,10 +472,161 @@ public FileModel createContent(FileModel fileModel) throws DataPreparationExcept
+     * Return the {@link Document} object on success creation
+     * <code>
+     * dataContent.usingUser(userModel).usingResource(myFolder).createContent(DocumentType.TEXT_PLAIN);
+     * </code>
+     *
+     * @param client
+     * @param fileModel
+     * @param documentType
+     * @return
+     * @throws DataPreparationException
+     */
+    public FileModel createContentDocTypeV1Api(AlfrescoHttpClient client, FileModel fileModel, DocumentType documentType)
+    {
+        // Build request
+        String nodeId = this.getNodeRef();
+        String reqUrl = client.getApiVersionUrl() + "nodes/" + nodeId + "/children";
+        String newContent = String.format("%s.%s", fileModel.getName(), Utility.cmisDocTypeToExtentions(documentType));
+
+        HttpPost post  = new HttpPost(reqUrl);
+        JSONObject body = new JSONObject();
+        body.put("name", newContent);
+        body.put("nodeType", "cm:content");
+        // Set Title or Description if specified
+        if (fileModel.getTitle() != null || fileModel.getDescription() != null)
+        {
+            body.put("cm:title", fileModel.getTitle());
+            body.put("cm:description", fileModel.getDescription());
+        }
+
+        post.setEntity(client.setMessageBody(body));
+
+        // Send Request
+        logger.info(String.format("Create content with name '%s' by: ", newContent));
+        logger.info(String.format("POST: '%s'", reqUrl));
+        HttpResponse response = client.execute(currentUser.getUsername(), currentUser.getPassword(), post);
+        if(HttpStatus.SC_CREATED == response.getStatusLine().getStatusCode())
+        {
+            JSONObject entryResponse = client.readStream(response.getEntity());
+            JSONObject entryValueMap = (JSONObject) entryResponse.get("entry");
+
+            fileModel.setNodeRef(entryValueMap.get("id").toString());
+            fileModel.setName(entryValueMap.get("name").toString());
+            fileModel.setFileType(FileType.fromName(fileModel.getName()));
+
+            String fileLocation = Utility.buildPath(getLastResource(), newContent);
+            fileModel.setCmisLocation(fileLocation);
+            fileModel.setProtocolLocation(fileLocation);
+            fileModel.setContent("This is a test file");
+
+            logger.info(String.format("Successful created content with id '%s' ", entryValueMap.get("id").toString()));
+            return fileModel;
+        }
+        else
+        {
+            logger.error(client.getParameterFromJSON(response,"briefSummary", "error"));
+            return new FileModel();
+        }
+    }
+
+    /**
+     * This method updates the content of the created document
+     *
+     * @param client
+     * @param fileModel
+     * @param documentType
+     */
+    public void updateContent(AlfrescoHttpClient client, FileModel fileModel, DocumentType documentType)
+    {
+        // Build request
+        String nodeId = fileModel.getNodeRef();

```

Comment:
error handling? what if fileModel does not represent a valid object/file/folder from the Repository ?